jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/LambdaRunConfigurationBase.kt

```diff
@@ -13,35 +12,24 @@ import com.intellij.openapi.fileEditor.FileDocumentManager
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.util.io.FileUtil
 import com.intellij.openapi.vfs.LocalFileSystem
-import com.intellij.util.xmlb.XmlSerializer
 import com.intellij.util.xmlb.annotations.Property
 import org.jdom.Element
 import software.aws.toolkits.core.credentials.CredentialProviderNotFound
 import software.aws.toolkits.jetbrains.core.credentials.CredentialManager
 import software.aws.toolkits.jetbrains.core.region.AwsRegionProvider
+import software.aws.toolkits.jetbrains.ui.connection.AwsConnectionsRunConfigurationBase
+import software.aws.toolkits.jetbrains.ui.connection.BaseAwsConnectionOptions
 import software.aws.toolkits.resources.message
 import java.nio.charset.StandardCharsets
 
 abstract class LambdaRunConfigurationBase<T : BaseLambdaOptions>(
     project: Project,
     private val configFactory: ConfigurationFactory,
     id: String
-) : LocatableConfigurationBase<T>(project, configFactory, id),
+) : AwsConnectionsRunConfigurationBase<T>(project, configFactory, id),
     RunConfigurationWithSuppressedDefaultRunAction,
     RunConfigurationWithSuppressedDefaultDebugAction {
 
-    protected abstract val lambdaOptions: BaseLambdaOptions
-
-    final override fun readExternal(element: Element) {
```

Comment:
Were the only settings getting persisted here the AWS Account settings?

Reply:
This persisted all the settings in the UI

Reply:
no - but this just uses bean-based serialization of the class - so any properties on the actual implementation should get persisted.

Reply:
Right so it seems weird to put this in the AWS Settings tab: we are persisting all settings through the AWS Settings tab?